mms/tests/unit_tests/test_validate_messages.py

```diff
@@ -0,0 +1,106 @@
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import pytest
+from mms.mxnet_model_service_error import MMSError
+from mms.utils.validators.validate_messages import ModelWorkerMessageValidators
+from mms.utils.model_server_error_codes import ModelServerErrorCodes
+
+def test_validate_load_message_missing_model_path():
+    invalid_msg = '{ \"modelName\" : \"some-model-name\" }'
+    with pytest.raises(MMSError) as error:
+        ModelWorkerMessageValidators.validate_load_message(invalid_msg)
+
+    assert error.value.get_code() == ModelServerErrorCodes.INVALID_LOAD_MESSAGE, 'error codes don\'t match'
+    assert error.value.get_message() == 'Load command missing \"model-path\" key'
+
+    valid_msg = '{ \"modelName\" : \"some-model-name\", \"modelPath\" : \"some-model-path\" }'
+    ModelWorkerMessageValidators.validate_load_message(valid_msg)
+
+def test_validate_load_message_missing_model_name():
+    invalid_msg = '{ \"modelPath\" : \"some-model-path\" }'
+    with pytest.raises(MMSError) as error:
+        ModelWorkerMessageValidators.validate_load_message(invalid_msg)
+
+    assert error.value.get_code() == ModelServerErrorCodes.INVALID_LOAD_MESSAGE, 'error codes don\'t match'
+    assert error.value.get_message() == 'Load command missing \"model-name\" key'
+
+    valid_msg = '{ \"modelName\" : \"some-model-name\", \"modelPath\" : \"some-model-path\" }'
+    ModelWorkerMessageValidators.validate_load_message(valid_msg)
+
+def test_validate_predict_data():
+    invalid_msg = '{\"request-id\" : \"some-key\"}'
+    with pytest.raises(MMSError) as error:
+        ModelWorkerMessageValidators.validate_predict_data(invalid_msg)
+
+    assert error.value.get_code() == ModelServerErrorCodes.INVALID_PREDICT_INPUT
+    assert error.value.get_message() == 'Predict command input missing \"request-id\" field'
+
+    valid_msg = '{ \"requestId\" : \"111-222-3333\"}'
+    ModelWorkerMessageValidators.validate_predict_data(valid_msg)
+
+def test_validate_predict_inputs_missing_value():
+    invalid_input = '{ \"input1\": { \"name\" : \"some-name\" } }'
+    with pytest.raises(MMSError) as error:
+        ModelWorkerMessageValidators.validate_predict_inputs(invalid_input)
+
+    assert error.value.get_code() == ModelServerErrorCodes.INVALID_PREDICT_INPUT
+    assert error.value.get_message() == 'Predict command input data missing \"value\" field'
+
+
+    valid_input = '{ \"input1\": {\"value\" : \"some-name\", \"name\" : \"some-name\"}}'
+    ModelWorkerMessageValidators.validate_predict_inputs(valid_input)
+
+def test_validate_predict_inputs_missing_name():
+    invalid_input = '{ \"input1\": { \"value\" : \"some-value\" } }'
```

Comment:
This particular test case might be wrong. I think the validator file is not updated with the request format. You might want to refer to "https://github.com/awslabs/mxnet-model-server/blob/netty/mms/model_service_worker.py#L215" for the correct request format. You have highlighted some issues with our current validation. You could add "modelInputs" to validator code as well, to validate "predict" message.

Reply:
Sure. Will check that out.

Reply:
Done in 3b280a1a8186c3a2e9a49c9d87f9320a26805ec2